### ISSUE TITLE
DeprecationWarning: generator '_sep_inserter' raised StopIteration - PEP 479

### ISSUE THREAD

**Issue by lykinsbd** [SethMMorton/natsort#42]
In Python 3.6.1, I'm experiencing an issue with a `DeprecationWarning` as shown below:

```
lib/python3.6/site-packages/natsort/utils.py:366: DeprecationWarning: generator '_sep_inserter' raised StopIteration
  return lambda split_val, val: tuple(split_val)
```

Code functions and proceeds, however it pops the above DeprecationWarning for every sort.

Appears to be [PEP 479](https://www.python.org/dev/peps/pep-0479/) related.  As of Python 3.5, `StopIteration` handling has changed inside generators.

From [PEP 479 section on the Transition Plan](https://www.python.org/dev/peps/pep-0479/#transition-plan), it seems like the correct timing as 3.6 is when the non-silent warning was instituted:

>     Python 3.5: Enable new semantics under __future__ import; silent deprecation warning if StopIteration bubbles out of a generator not under __future__ import.
>     Python 3.6: Non-silent deprecation warning.
>     Python 3.7: Enable new semantics everywhere.

The Natsort code itself shows a comment about `StopIteration` inside of the referenced `_sep_inserter` method:
https://github.com/SethMMorton/natsort/blob/master/natsort/utils.py#L233

```
# Get the first element. If StopIteration is raised, that's OK.
```

Seems like we may just need to handle the raised `StopIteration` exception as outlined in PEP 479 and execute inside a `try...except` block to catch `StopIteration`.

**Comment by SethMMorton** [comment]
Thanks for reporting this!  Would you be interested in making a pull request to fix this?

**Comment by lykinsbd** [comment]
Yeah, I'd be happy to.  I was actually planning on it, just wanted to document the issue itself before I forgot about it.

I'll take a swing at it this weekend.

**Comment by SethMMorton** [comment]
Any update on this?

**Comment by lykinsbd** [comment]
@SethMMorton I put a fix in that seems to take care of this.

**Comment by SethMMorton** [comment]
This was fixed with merge request #43. I will close the issue when then next version of `natsort` is released.

**Comment by SethMMorton** [comment]
Added in version 5.1.0.
